### ISSUE TITLE
UI Line Renderer with relative mode crashes Unity, when adding large number

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#455]
<sup>Issue created by Luca Heft as [Bitbucket Issue #&#8203;316](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/316) on 2020.03.16 08:33.</sup>
This bug happened to me when i had a parsing error, which resulted in adding a large number to the UILineRenderer.  
  
This will crash the editor and build. Error in the editor log will say something allong the lines of:  
Dynamic array capacity overflow

I attached a sample project. Unfortunetly i cant reproduce the error message with it and the editor just crashes.

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.03.16 10:17](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/316#comment-56535816), @SimonDarksideJ commented:</sup>
Thanks for the sample and I’ll look at that right away. I have a few tricks for handling those pesky editor crashes.

One you can do yourself is simply to check the Unity editor log, located in

```
C:\Users\<username>\AppData\Local\Unity\Editor
```

Just start at the end and work your way back to the exception.  
  
P.S.

When sharing a repo or project, only send the Assets and possibly the Project Settings / Packages folders.  Never send Library / obj / bin and Temp :smiley: they aren’t needed and bloat a zip.

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.03.16 10:48](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/316#comment-56536312), @SimonDarksideJ commented:</sup>
Very interesting @{5e5e154208e2a40c9386425c} , this is likely a Unity bug I suspect given Unity doesn’t actually crash, it simply goes in to an infernal loop inside the engine.  Going to continue testing.

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.03.16 12:01](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/316#comment-56537289), @SimonDarksideJ commented:</sup>
SO, reached a good news / bad news situation here and surprisingly, Unity is NOT crashing.

Been running through many simulations and tests to eventually reach a conclusion, and the actual issue is you are hitting the extreme limits of Unity’s 3D rendering distance.

![](https://bitbucket.org/repo/RXA7eg/images/3380578073-BF88D0EC-A4F8-4348-A8AF-E0A266F72F11.GIF)
If you watch carefully, sorry for the quality, but at the upper bounds of E\+13 \(it starts around E\+10\) but Unity’s rendering actually slows down as it attempts to render beyond its normal bounds :smiley: . This is very interesting as I’ve never come across it before. I can only surmise that Unity is having to extend it’s own rendering bounds/queues in an attempt to create a 3D world \(like cubic\) large enough to include the render data.

‌

In short, normalise your graph and try to keep it within the E\+5 limit at most, else you will see a severe degradation of rendering performance, likely more so on slower platforms :smiley:

‌

Hope that helps.

P.S.  
I reduced your AddPoints method to this using a list

```
    List<Vector2> _points = new List<Vector2>();

    public void AddPointNormal(float value)
    {
        _points.Add(new Vector2(1, value));
        lineRenderer.Points = _points.ToArray();
        lineRenderer.SetAllDirty();

    }
```

Although it’s still better to use a fixed array and a set outer bound \(Vector2\[20000\]\) and just reuse elements within it for the best performance.

‌

Hope that helps @{5e5e154208e2a40c9386425c}

‌

P.S.

Close the issue if you are ok with the resolution

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.03.16 12:25](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/316#comment-56537635), Luca Heft commented:</sup>
Sounds good to me. I fixed my parsing error, so this wont happen any more. Thanks for investigation and detailed explanation! :smiley:

**Comment by SimonDarksideJ** [comment]
No response, expect closed
